packages/shopify-codemod/transforms/simplify-?=-if-else-statement.js

```diff
@@ -0,0 +1,62 @@
+export default function simplifyCondAssignIfElseStatement({source}, {jscodeshift: j}, {printOptions = {quote: 'single'}}) {
+
+  function testIsBinaryExpression(path) {
+    return j.BinaryExpression.check(path.get('test').node);
+  }
+
+  function testChecksIdentifierForNotNull(path) {
+    const test = path.get('test').node;
+    return (
+      j.Identifier.check(test.left) &&
+      test.operator === '!=' &&
+      j.Literal.check(test.right) &&
+      test.right.value === null
+    );
+  }
+
+  function representsConditionalAssignment(path) {
+    const primaryIdentifierName = path.get('test', 'left', 'name').value;
+    return (
+      isBlockStatementContainingOnlyExpressionStatement(path.get('consequent')) &&
+      isBlockStatementContainingOnlyExpressionStatement(path.get('alternate')) &&
+      consequentIsPrimaryIdentifier(path, primaryIdentifierName) &&
+      alternateAssignsPrimaryIdentifier(path, primaryIdentifierName)
+    );
+  }
+
+  function isBlockStatementContainingOnlyExpressionStatement({node}) {
+    return (
+      j.BlockStatement.check(node) &&
+      node.body.length === 1 &&
+      j.ExpressionStatement.check(node.body[0])
+    );
+  }
+
+  function consequentIsPrimaryIdentifier(path, primaryIdentifierName) {
+    const consequentExpression = path.get('consequent', 'body', 0, 'expression');
+    return (
+      j.Identifier.check(consequentExpression.node) &&
+      consequentExpression.get('name').value === primaryIdentifierName
+    );
+  }
+
+  function alternateAssignsPrimaryIdentifier(path, primaryIdentifierName) {
+    const alternateExpression = path.get('alternate', 'body', 0, 'expression');
+    return (
+      j.AssignmentExpression.check(alternateExpression.node) &&
+      alternateExpression.get('left', 'name').value === primaryIdentifierName
+    );
+  }
+
+  return j(source)
```

Comment:
So there's a (totally not documented) structure that our transforms mostly follow:
1. Use `find` to filter by *structure*
2. Use `filter` to filter by *content*
3. Transform with the assumption that 1+2 have taken care of edge cases

I think you could lean more on the first step.  It's commendable that you place great emphasis on creating readable, flowing transforms *but* you can also assume that your reader has a degree of familiarity with `jscodeshift`.

The following is reasonably grokable by someone with a few hours of experience, and is similar in level-of-effort once you account for your eyes being able to stay in one spot.

```
    .find(j.IfStatement, {
      test: {
        type: j.BinaryExpression.name,
        left: {type: j.Identifier.name},
        operator: '!=',
        right: {type: j.Literal.name, value: null},
      },
      consequent: {
        body: (body) => body.length === 1 && j.Identifier.check(body[0].expression),
      },
      alternate: {
        body: (body) => body.length === 1 && j.AssignmentExpression.check(body[0].expression),
      },
    })
    .filter(({node}) => {
      const testId = node.test.left.name;
      const consequentId = node.consequent.body[0].expression.name;
      const alternateId  = node.alternate.body[0].expression.left.name;

      return testId === consequentId && testId === alternateId;
    })
```